pyomo/contrib/pynumero/algorithms/solvers/pyomo_ext_cyipopt.py

```diff
@@ -286,6 +286,10 @@ def __init__(
 
         # currently, this interface does not do anything with Hessians
 
+        # Call CyIpoptProblemInterface.__init__, which calls
+        # cyipopt.Problem.__init__
+        super(PyomoExternalCyIpoptProblem, self).__init__()
+
```

Comment:
Note that this `__init__` call, which wasn't previously necessary, is now necessary in subclasses of `CyIpoptProblemInterface` to make sure that the `cyipopt.Problem` get initialized so we don't segfault when we try to solve it later.

In this way, this PR could break some user code if they are subclassing `CyIpoptProblemInterface`.

Reply:
I am thinking of adding an `__initialized` flag to `CyIpoptProblemInterface.__init__` so we can raise a warning/error if `solve` is called when `cyipopt.Problem` hasn't been initialized.